src/Stratis.FederatedPeg.Features.FederationGateway/Interfaces/IDepositExtractor.cs

```diff
@@ -11,12 +11,37 @@ namespace Stratis.FederatedPeg.Features.FederationGateway.Interfaces
     /// </summary>
     public interface IDepositExtractor
     {
+        /// <summary>
+        /// Extracts deposits from the transactions in a block.
+        /// </summary>
+        /// <param name="block">The block containing the transactions.</param>
+        /// <param name="blockHeight">The height of the block containing the transactions.</param>
+        /// <returns>The extracted deposit (if any), otherwise returns an empty list.</returns>
         IReadOnlyList<IDeposit> ExtractDepositsFromBlock(Block block, int blockHeight);
-      
+
+        /// <summary>
+        /// Extracts a deposit from a transaction (if any).
+        /// </summary>
+        /// <param name="transaction">The transaction to extract deposits from.</param>
+        /// <param name="blockHeight">The block height of the block containing the transaction.</param>
+        /// <param name="blockHash">The block hash of the block containing the transaction.</param>
+        /// <returns>The extracted deposit (if any), otherwise <c>null</c>.</returns>
         IDeposit ExtractDepositFromTransaction(Transaction transaction, int blockHeight, uint256 blockHash);
 
+        /// <summary>
```

Comment:
so if I understand, we need both
1) ExtractMaturedBlockDeposits => from a Source chain's point of view (A chain)
2) ExtractBlockDeposits => from a Target node's point of view (B chain)
correct ?

Reply:
The ExtractBlockDeposits makes a lot more sense from Node B's perspective, or when extracting more than  one block at a time.